pandas/tests/io/formats/test_format.py

```diff
@@ -305,16 +305,14 @@ def test_repr_non_interactive(self):
             assert not has_truncated_repr(df)
             assert not has_expanded_repr(df)
 
-    def test_repr_truncates_terminal_size(self, mock):
-        # https://github.com/pandas-dev/pandas/issues/21180
-        # TODO: use mock fixutre.
-        # This is being backported, so doing it directly here.
+    def test_repr_truncates_terminal_size(self, monkeypatch):
+        # see gh-21180
 
         terminal_size = (118, 96)
-        p1 = mock.patch('pandas.io.formats.console.get_terminal_size',
-                        return_value=terminal_size)
-        p2 = mock.patch('pandas.io.formats.format.get_terminal_size',
-                        return_value=terminal_size)
+        monkeypatch.setattr('pandas.io.formats.console.get_terminal_size',
```

Comment:
Does this reset the attribute after the test or do we need to use the context manager?

Reply:
@WillAyd the context manager is only required when patching might break `pytest` or where it is required to undo a patch before the test ends. the `monkeypatch` fixture will undo all patches on test teardown.